### PR TITLE
#2189 Add more e2e test with perf connector and metrics

### PR DESCRIPTION
**Pull request by wu87988622**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #2189 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

**Comment by wu87988622** [comment]
retry manager-e2e

**Comment by wu87988622** [comment]
retry manager-e2e

**Comment by wu87988622** [comment]
retry manager-e2e

**Comment by wu87988622** [comment]
retry manager-e2e

**Comment by wu87988622** [comment]
retry manager-e2e

File: ohara-manager/client/cypress.json [hunk 1]
```diff
@@ -9,5 +9,7 @@
   "video": false,
   "requestTimeout": 30000,
   "defaultCommandTimeout": 20000,
-  "watchForFileChanges": false
+  "watchForFileChanges": false,
+  "viewportWidth": 1920,
```
Comment (konekoya) [inline ohara-manager/client/cypress.json]: Why we're adding these two configs?

File: ohara-manager/client/cypress/e2e/pipelineNewPage.test.js [hunk 2]
```diff
@@ -332,4 +332,75 @@ describe('PipelineNewPage', () => {
       .get('g.edgePath')
       .should('have.length', 2);
   });
+
+  it('connects perf source -> Topic check metrics', () => {
```
Comment (konekoya) [inline ohara-manager/client/cypress/e2e/pipelineNewPage.test.js]:
```suggestion
  it('connects perf source -> Topic', () => {
```

File: ohara-manager/client/cypress/e2e/pipelineNewPage.test.js [hunk 3]
```diff
@@ -332,4 +332,75 @@ describe('PipelineNewPage', () => {
...
+    cy.server();
+    cy.route('PUT', '/api/pipelines/*').as('putPipeline');
+    cy.route('GET', '/api/pipelines/*').as('getPipeline');
+    cy.route('GET', '/api/connectors/*').as('getConnector');
+
+    const perfName = makeRandomStr();
+    const topicName = Cypress.env('TOPICS_NAME');
```
Comment (konekoya) [inline ohara-manager/client/cypress/e2e/pipelineNewPage.test.js]: Does this `topicName` container only a topic name or multiple ones?

File: ohara-manager/client/cypress/e2e/pipelineNewPage.test.js [hunk 4]
```diff
@@ -332,4 +332,75 @@ describe('PipelineNewPage', () => {
...
+      .click()
+      .getByText('Please select...')
+      .click()
+      .get(`li[data-value=${topicName}]`)
+      .click()
+      .wait(2000)
+      .wait('@putPipeline')
+      .getByTestId('start-btn')
+      .click({ force: true })
+      .wait('@getPipeline') //waiting for metrics make sure graph have metrics
```
Comment (konekoya) [inline ohara-manager/client/cypress/e2e/pipelineNewPage.test.js]:
```suggestion
      .wait('@getPipeline') // Waiting for multiple `GET pipeline` requests here, so the metrics data can be ready for us to test
```

File: ohara-manager/client/cypress/e2e/pipelineNewPage.test.js [hunk 5]
```diff
@@ -332,4 +332,75 @@ describe('PipelineNewPage', () => {
...
+      .click({ force: true })
+      .getByText(`Metrics (${topicName})`)
+      .should('have.length', 1)
+      .getByTestId('expandIcon')
+      .click({ force: true })
+      .get('@getPipeline')
+      .then(xhr => {
+        const topic = xhr.response.body.objects[0];
+        const metricsCount = topic.metrics.meters.length;
+        cy.get('.item').should('have.length', metricsCount);
```
Comment (konekoya) [inline ohara-manager/client/cypress/e2e/pipelineNewPage.test.js]: Hmm, we should avoid getting element by CSS class in general. Maybe add a `data-testid` to these items? CSS classes are meant to used for the styling purpose. So we won't break the test when we change these classes due to edit/remove CSS classes

File: ohara-manager/client/cypress/e2e/pipelineNewPage.test.js (hunk 3 above)
Comment (konekoya) [reply ohara-manager/client/cypress/e2e/pipelineNewPage.test.js]: I just want to make sure if it's a single topic name, shouldn't the env name be `TOPIC_NAME`?

File: ohara-manager/client/cypress.json (hunk 1 above)
Comment (wu87988622) [reply ohara-manager/client/cypress.json]: The test has failed. I found that the difference between ci and gui is the size of the window. I am still looking for a better size.

File: ohara-manager/client/cypress/e2e/pipelineNewPage.test.js (hunk 3 above)
Comment (wu87988622) [reply ohara-manager/client/cypress/e2e/pipelineNewPage.test.js]: The topic will be created before this test, and the topic will be used in this test.

File: ohara-manager/client/cypress.json (hunk 1 above)
Comment (konekoya) [reply ohara-manager/client/cypress.json]: Cool, if that's the case. But I think we should use or app size?

File: ohara-manager/client/cypress/e2e/pipelineNewPage.test.js (hunk 3 above)
Comment (konekoya) [reply ohara-manager/client/cypress/e2e/pipelineNewPage.test.js]: Yep, I know that. But is it singular or plural?

**Comment by wu87988622** [comment]
retry manager-e2e

File: ohara-manager/client/cypress/e2e/pipelineNewPage.test.js (hunk 3 above)
Comment (wu87988622) [reply ohara-manager/client/cypress/e2e/pipelineNewPage.test.js]: oh i know what are you i will change to TOPIC_NAME

File: ohara-manager/client/cypress/e2e/pipelineNewPage.test.js (hunk 2 above)
Comment (wu87988622) [reply ohara-manager/client/cypress/e2e/pipelineNewPage.test.js]: done

File: ohara-manager/client/cypress/e2e/pipelineNewPage.test.js (hunk 4 above)
Comment (wu87988622) [reply ohara-manager/client/cypress/e2e/pipelineNewPage.test.js]: done

File: ohara-manager/client/cypress/e2e/pipelineNewPage.test.js (hunk 5 above)
Comment (wu87988622) [reply ohara-manager/client/cypress/e2e/pipelineNewPage.test.js]: ok done

File: ohara-manager/client/cypress.json (hunk 1 above)
Comment (wu87988622) [reply ohara-manager/client/cypress.json]: viewportWidth 1200, because we web design is 1200

**Comment by wu87988622** [comment]
retry manager-e2e

File: ohara-manager/client/cypress/e2e/pipelineNewPage.test.js [hunk 6]
```diff
@@ -325,11 +323,83 @@ describe('PipelineNewPage', () => {
...
-      .get(`li[data-value=${Cypress.env('TOPICS_NAME')}]`)
+      .get(`li[data-value=${Cypress.env('TOPIC_NAME')}]`)
       .click()
       .wait(2000) // UI has one sec throttle, so we need to wait a bit time and then wait for the request
       .wait('@putPipeline')
       .get('g.edgePath')
       .should('have.length', 2);
   });
+
+  it.only('connects perf source -> Topic', () => {
```
Comment (konekoya) [inline ohara-manager/client/cypress/e2e/pipelineNewPage.test.js]: Please remember to remove the `only`

**Comment by wu87988622** [comment]
retry manager-e2e

File: ohara-manager/client/cypress/e2e/pipelineNewPage.test.js (hunk 6 above)
Comment (wu87988622) [reply ohara-manager/client/cypress/e2e/pipelineNewPage.test.js]: ok

**Comment by wu87988622** [comment]
retry manager-e2e

**Comment by wu87988622** [comment]
retry manager-e2e

**Comment by wu87988622** [comment]
retry manager-e2e

**Comment by chia7712** [comment]
Our IT node seems to be dead. Please don’t retry it now :(

**Comment by wu87988622** [comment]
> Our IT node seems to be dead. Please don’t retry it now :(

ok~ thanks.

**Comment by wu87988622** [comment]
retry manager-e2e

**Comment by wu87988622** [comment]
retry manager-e2e

**Comment by wu87988622** [comment]
retry manager-e2e

**Comment by wu87988622** [comment]
retry manager-e2e

**Comment by wu87988622** [comment]
retry manager-e2e

**Comment by wu87988622** [comment]
retry manager-e2e

**Comment by wu87988622** [comment]
unrelated fail

**Pull request merged**